File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/plannedtransaction/PlannedTransType.java
package fpt.aptech.server.enums.plannedtransaction;

import lombok.Getter;

/**
 * Phân loại giao dịch sẽ được sinh ra từ một giao dịch dự kiến.
 * Dùng cho tPlannedTransactions.trans_type
 */
@Getter
public enum PlannedTransType {
    EXPENSE(1),         // Khoản chi
    INCOME(2),          // Khoản thu
    LOAN_OUT(3),        // Cho vay
    LOAN_IN(4),         // Đi vay
    DEBT_COLLECT(5),    // Thu nợ
    DEBT_REPAY(6);      // Trả nợ

    private final int value;

    PlannedTransType(int value) {
        this.value = value;
    }
}

File: APP MOBILE/smart_money/lib/modules/planned/enums/planned_trans_type.dart
/// Loại giao dịch từ kế hoạch: Chi, Thu, Cho vay, Đi vay, Thu nợ, Trả nợ
enum PlannedTransType {
  expense,      // 1 - Chi
  income,       // 2 - Thu
  loanOut,      // 3 - Cho vay
  loanIn,       // 4 - Đi vay
  debtCollect,  // 5 - Thu nợ
  debtRepay;    // 6 - Trả nợ

  int get value => index + 1;

  static PlannedTransType fromValue(int value) {
    if (value < 1 || value > 6) {
      return PlannedTransType.expense;
    }
    return PlannedTransType.values[value - 1];
  }

  String get displayName {
    switch (this) {
      case PlannedTransType.expense:
        return 'Expense';
      case PlannedTransType.income:
        return 'Income';
      case PlannedTransType.loanOut:
        return 'Lend';
      case PlannedTransType.loanIn:
        return 'Borrow';
      case PlannedTransType.debtCollect:
        return 'Debt collection';
      case PlannedTransType.debtRepay:
        return 'Debt repayment';
    }
  }
}

